### PR TITLE
MGMT-15570: Enable BMC form in Add hosts modal, for all platforms, if provisioning configuration exists

### PR DESCRIPTION
**Pull request by batzionb**
Provisioning configuration exists - BMC options are enabled with no help icon

No provisioning configuration - BMC options are disabled and there's a help icon explaining why and links to console default create page of Provisioning configuration

image

Provisioning configuration loading - BMC options are disabled and showing spinner

Peek 2023-09-04 16-26

Error loading Provisioning configuration - The BMC options are enabled and the error shows at the top of the modal
image

### CODE REVIEW

**Review by ammont82 — APPROVED** [review]

File: libs/ui-lib/lib/cim/components/InfraEnv/AddHostDropdown.tsx [hunk 1]
```diff
@@ -24,10 +45,12 @@ const AddHostDropdown = ({
   onCreateBMH,
   docVersion,
   onCreateBmcByYaml,
-  isBMPlatform,
+  provisioningConfigResult,
 }: AddHostDropdownProps) => {
   const [addModalType, setAddModalType] = React.useState<ModalType>(undefined);
   const [isKebabOpen, setIsKebabOpen] = React.useState(false);
+  const [provisioningConfig, provisioningConfigLoaded, provisioningConfigError] =
+    provisioningConfigResult || [null, true, undefined];
```
Comment (rawagner) [inline libs/ui-lib/lib/cim/components/InfraEnv/AddHostDropdown.tsx:53]: why is this needed ? the type of `provisioningConfigResult` does not allow for undefined/null

File: libs/ui-lib/lib/cim/components/modals/ProvisioningConfigErrorAlert.tsx [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+import { useTranslation } from '../../../common/hooks/use-translation-wrapper';
+
+const ProvisioningConfigErrorAlert = ({ error }: { error: unknown }) => {
+  const { t } = useTranslation();
+  if (!error) {
+    return null;
+  }
+  return (
+    <Alert title={t('ai:Failed to get Provisioning Configuration')} variant="danger" isInline>
+      {(error as Error).message || ''}
```
Comment (rawagner) [inline libs/ui-lib/lib/cim/components/modals/ProvisioningConfigErrorAlert.tsx:12]: to get the err msg, you could use `getErrorMessage` from `common/utils`

**Review by ammont82 — APPROVED** [review]

**Review by rawagner — APPROVED** [review]

File: libs/ui-lib/lib/cim/components/modals/ProvisioningConfigErrorAlert.tsx (hunk 2 above)
Comment (batzionb) [reply libs/ui-lib/lib/cim/components/modals/ProvisioningConfigErrorAlert.tsx]: fixed
